src/Common/src/System/Security/Cryptography/ECDsaOpenSsl.ImportExport.cs

```diff
@@ -105,12 +105,6 @@ private static ECParameters ExportNamedCurveParameters(SafeEcKeyHandle key, bool
 
                 ECParameters parameters = Interop.Crypto.GetECKeyParameters(key, includePrivateParameters);
 
-                bool hasPrivateKey = (parameters.D != null);
-                if (hasPrivateKey != includePrivateParameters)
-                {
-                    throw new CryptographicException(SR.Cryptography_CSP_NoPrivateKey);
-                }
-
```

Comment:
Leaving these for defense in depth seems good.

Reply:
Sure. They'll be unreachable but will protect against code changes in the future.